packages/devtools_app/lib/src/preferences.dart

```diff
@@ -60,4 +68,34 @@ class PreferencesController {
   void toggleDenseMode(bool enableDenseMode) {
     _denseMode.value = enableDenseMode;
   }
+
+  /// Get the value for the split fractions.
+  ValueNotifier<List<double>> lookupSplitFractions(String key) {
+    return splitFractions.putIfAbsent(
+        key, () => ValueNotifier<List<double>>(_initialFractions[key] ?? []));
+  }
+
+  /// Update the value for the split fractions.
+  void updateSplitFractions(String key, List<double> value) {
+    if ((key?.isEmpty ?? true) ||
+        (value?.length ?? 0) < 2 ||
+        isListEqual(lookupSplitFractions(key).value, value)) return;
+    lookupSplitFractions(key).value = value;
+    storage?.setValue(
+        'ui.splitFractions', jsonEncode(fractionsToJson(splitFractions)));
+  }
+
+  Map<String, List<double>> fractionsFromJson(Map<String, dynamic> json) {
+    return json
+        .map((key, fractions) => MapEntry(key, fractions?.cast<double>()));
+  }
+
+  Map<String, List<double>> fractionsToJson(
+      Map<String, ValueListenable<List<double>>> fractions) {
+    return fractions.map((key, fractions) => MapEntry(key, fractions.value));
+  }
+
+  bool isListEqual(List value, List other) {
```

Comment:
add the `<double>` type to these lists

Reply:
additionally, return early `if ((value == null) != (other == null))` to avoid unnecessary encoding. Since we expect these lists to be very small, it may be more performant to use [ListEquality](https://api.flutter.dev/flutter/package-collection_collection/ListEquality-class.html) to compare the elements directly instead of encoding the lists.